src/protocol/uniswap_v3/state.rs

```diff
@@ -279,13 +280,18 @@ impl ProtocolSim for UniswapV3State {
         let result = self.swap(zero_for_one, amount_specified, None)?;
 
         trace!(?amount_in, ?token_a, ?token_b, ?zero_for_one, ?result, "V3 SWAP");
+        let mut new_state = self.clone();
+        new_state.liquidity = result.liquidity;
+        new_state.tick = result.tick;
+        new_state.sqrt_price = result.sqrt_price;
```

Comment:
I _think_ this is enough but can any uniswap v3 expert check it please?

Reply:
don't you need to update the tick list?

Reply:
I don't really know. but it isn't available in the result..

Reply:
Update - it's not needed. The tick list does not change between swaps.